SwiftCompilerSources/Sources/Optimizer/ModulePasses/MandatoryPerformanceOptimizations.swift

```diff
@@ -75,6 +75,12 @@ private func optimize(function: Function, _ context: FunctionPassContext) {
       switch instruction {
       case let apply as FullApplySite:
         inlineAndDevirtualize(apply: apply, alreadyInlinedFunctions: &alreadyInlinedFunctions, context, simplifyCtxt)
+      case let alloc as AllocRefInst:
+        specializeVTableAndAddEntriesToWorklist(for: alloc.type, in: function, context, &worklist)
```

Comment:
cosmetic: IMO, it would be clearer to pull the `hasFeature(Feature::Embedded)` check here, e.g. `if context.isEmbedded { ... }`. Otherwise the function name should be something (ugly) like `specializeVTableAndAddEntriesToWorklistIfNotEmbedded`.
At least, you should add a comment. But expressing a comment as (readable) code is the better option :-)

Reply:
Done!